src/odemis/gui/model/__init__.py

```diff
@@ -1606,17 +1606,16 @@ def _getStageLimitsXY(self):
         stage_limits = {}
         # Physical stage limits
         if "x" in self._stage.axes and hasattr(self._stage.axes["x"], "range"):
-            stage_limits.update({"x": [self._stage.axes["x"].range[0], self._stage.axes["x"].range[1]]})
+            stage_limits["x"] = list(self._stage.axes["x"].range)
         if "y" in self._stage.axes and hasattr(self._stage.axes["y"], "range"):
-            stage_limits.update({"y": [self._stage.axes["y"].range[0], self._stage.axes["y"].range[1]]})
+            stage_limits["y"] = list(self._stage.axes["x"].range)
```

Comment:
After reading again the code, I realise that `move` always contains x and y. In general, a stage without x and y is not supported. So you can just assume that x and y are present on the stage as well. It'll simplify a bit the code.